File: 268. Missing Number/268. Missing Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function compare(a, b) {
    return a - b;
}

var missingNumber = function(nums) {
    nums.sort(compare);
    for (var i = 0; i < nums.length; i++) {
        if (nums[i] != i) {
            return i;
        }
    }
    return nums.length;
};

File: 268. Missing Number/268. Missing Number.c
int compInc(const void *a, const void *b) {
    return *(int *)a - *(int *)b;    
}

int missingNumber(int* nums, int numsSize) {
    qsort(nums, numsSize, sizeof(nums[0]), compInc);
    int i = 0;
    for (i = 0; i < numsSize; i++) {
        if (nums[i] != i){
            return i;
        }
    }
    return numsSize;  // final value is numsSize
}
